packages/neon-core/src/sc/ContractParam.ts

```diff
@@ -16,16 +16,31 @@ export enum ContractParamType {
   Signature = 0x17, // TODO: Implement support
 
   Array = 0x20,
-  Map = 0x22, // TODO: Implement support
+  Map = 0x22,
 
   InteropInterface = 0x30, // TODO: Implement support
 
   Void = 0xff,
 }
+export type ContractParamMapLike = {
+  key: ContractParamLike;
+  value: ContractParamLike;
+}[];
+
+export type ContractParamMap = {
+  key: ContractParam;

```

Comment:
on a side note, does neo actually allow anything to be the key? Can I put a map as the key?

Reply:
No you can't. Only `string`, `bool` and `int` can be used as key type (what they call `PrimitiveType` in the VM).

Reply:
@snowypowers Should I create a validation for it ?

Reply:
If we can constraint the type, then yes please. Else, at least some comments would help out.